### PR TITLE
J unit tests iter1

### PR DESCRIPTION
**Pull request by Shashaank5**
J unit tests development for the first iteration

### CODE REVIEW

File: src/FloorSubsystem/FloorTest.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package FloorSubsystem;
+
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+
+class FloorTest {
+
```
Comment (Justgo13) [inline src/FloorSubsystem/FloorTest.java:7]: look into the setup() and teardown() methods, they are usually used for JUnit

File: src/SchedulerSubsystem/SchedulerDataGramCommunicatorTest.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+/**
+ * 
+ * @author Shashaank
+ * 		Testing 
+ *
+ */
+class SchedulerDataGramCommunicatorTest {
+
+	@Test
+	void testElevatorToFloor() {
```
Comment (Justgo13) [inline src/SchedulerSubsystem/SchedulerDataGramCommunicatorTest.java]: this test case name doesn't make sense, it looks like what is being tested in the flow where floor is producer and elevator is consumer.

File: src/SchedulerSubsystem/SchedulerDataGramCommunicatorTest.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+		
+		byte[] packet = testString.getBytes();
+		dsg.sendToElevator(packet);
+		
+		byte[] returned = dsg.receiveFromFloor();
+		Assert.assertEquals(packet, returned);
+	}
+	
+	@Test
+	void testFloorToElevator() {
```
Comment (Justgo13) [inline src/SchedulerSubsystem/SchedulerDataGramCommunicatorTest.java]: same with this one, it is seems like it should be testElevatorToFloor

**Review by Justgo13 — CHANGES_REQUESTED** [review]

File: src/SchedulerSubsystem/SchedulerDataGramCommunicatorTest.java (hunk 2 above)
Comment (Shashaank5) [reply src/SchedulerSubsystem/SchedulerDataGramCommunicatorTest.java]: Fixed this and renamed methods in datagramcommunicator

File: src/FloorSubsystem/FloorTest.java (hunk 1 above)
Comment (Shashaank5) [reply src/FloorSubsystem/FloorTest.java:7]: Don't think this is necessary as of yet but will implement if test grow larger and all require same instantiations.

File: src/FloorSubsystem/FloorSystem.java [hunk 4]
```diff
@@ -45,10 +46,10 @@ public void run() {
 		//Assume that for iteration 1, each message sent by the floor will eventually be received again
 		for(int i = 0; i < floorDataEntry.size(); i++) {
 			System.out.println("Sending message from Floor System to Scheduler.");
-			sharedCommunicator.sendToElevator(floorDataEntry.get(i));
+			sharedCommunicator.elevatorToFloorPut(floorDataEntry.get(i));
```
Comment (Justgo13) [inline src/FloorSubsystem/FloorSystem.java]: shouldn't htis be floorToElevatorPut

File: src/FloorSubsystem/FloorSystem.java [hunk 5]
```diff
@@ -45,10 +46,10 @@ public void run() {
...
 			
 			try {
-				System.out.println("Floor System received message from Scheduler: \n" + SerializeUtils.deserialize(sharedCommunicator.receiveFromElevator()));
+				System.out.println("Floor System received message from Scheduler: \n" + SerializeUtils.deserialize(sharedCommunicator.floorToElevatorGet()));
```
Comment (Justgo13) [inline src/FloorSubsystem/FloorSystem.java]: and this elevatorToFloorGet

File: src/ElevatorSubsystem/ElevatorSystem.java [hunk 6]
```diff
@@ -29,16 +29,14 @@ public ElevatorSystem(SchedulerDataGramCommunicator communicator) {
...
-			byte[] message = communicator.receiveFromFloor();
-			try {
-				System.out.println("Elevator System received message from Scheduler: \n" + SerializeUtils.deserialize(message));
-			} catch (ClassNotFoundException | IOException e) {
-				e.printStackTrace();
-			}
-			System.out.println("Sending message from Elevator System to Scheduler.");
-			communicator.sendToFloor(message);			
+		//assumes that only one message is sent, which is to be sent back
+		byte[] message = communicator.elevatorToFloorGet();
```
Comment (Justgo13) [inline src/ElevatorSubsystem/ElevatorSystem.java]: i think this should be floorToElevatorGet

File: src/ElevatorSubsystem/ElevatorSystem.java [hunk 7]
```diff
@@ -29,16 +29,14 @@ public ElevatorSystem(SchedulerDataGramCommunicator communicator) {
...
+		try {
+			System.out.println("Elevator System received message from Scheduler: \n" + SerializeUtils.deserialize(message));
+		} catch (ClassNotFoundException | IOException e) {
+			e.printStackTrace();
 		}
+		System.out.println("Sending message from Elevator System to Scheduler.");
+		communicator.floorToElevatorPut(message);
```
Comment (Justgo13) [inline src/ElevatorSubsystem/ElevatorSystem.java]: and this elevatorToFloorPut

**Review by Justgo13 — APPROVED** [review]

**Pull request merged**